File: framework/src/gmsec5/field/CharField.cpp
/*
 * Copyright 2007-2023 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


#include <gmsec5/field/CharField.h>
#include <gmsec5/internal/field/InternalCharField.h>


using namespace gmsec::api5;
using namespace gmsec::api5::internal;


CharField::CharField(const char* name, GMSEC_CHAR value, bool isHeader)
	: m_internal(new InternalCharField(name, value, isHeader))
{
	registerChild(m_internal);
}


CharField::CharField(const CharField& other)
	: m_internal(new InternalCharField(other.getName(), other.getValue(), other.isHeader()))
{
	registerChild(m_internal);
}



CharField::~CharField()
{
	delete m_internal;
}


GMSEC_CHAR CharField::getValue() const
{
	return m_internal->getValue();
}


const char* CharField::toXML() const
{
	return m_internal->toXML();
}


const char* CharField::toJSON() const
{
	return m_internal->toJSON();
}

File: java/gov/nasa/gsfc/gmsec/api5/field/CharField.java
/*
 * Copyright 2007-2024 United States Government as represented by the
 * Administrator of The National Aeronautics and Space Administration.
 * No copyright is claimed in the United States under Title 17, U.S. Code.
 * All Rights Reserved.
 */


/**
 * @file CharField.java
 */

package gov.nasa.gsfc.gmsec.api5.field;

import gov.nasa.gsfc.gmsec.api5.GmsecException;
import gov.nasa.gsfc.gmsec.api5.jni.field.JNICharField;


/**
 * Specialized Field class that can be used to retain a single character value.
 */
public class CharField extends Field
{
	//! @cond
	/**
	 * @hidden
	 * This constructor is for internal GMSEC API use only.
	 * @param field A JNICharField object.
	 */
	public CharField(JNICharField field)
	{
		setInternal(field);
	}
	//! @endcond


	/**
	 * Constructor.
	 *
	 * @param name Name of the field.
	 * @param data Character data to associate with the field.
	 *
	 * @throws IllegalArgumentException Thrown if the field name string is null, is an empty string, or is not compliant.
	 */
	public CharField(String name, char data) throws IllegalArgumentException
	{
		this(name, data, false);
	}


	/**
	 * Constructor.
	 *
	 * @param name Name of the field.
	 * @param data Character data to associate with the field.
	 * @param isHeader used to indicate if Field is a header field.
	 *
	 * @throws IllegalArgumentException Thrown if the field name string is null, is an empty string, or is not compliant.
	 */
	public CharField(String name, char data, boolean isHeader) throws IllegalArgumentException
	{
		validateName(name);

		setInternal(new JNICharField(name, data, isHeader));
	}


	/**
	 * Copy constructor.
	 *
	 * @param other The other CharField object to copy.
	 *
	 * @throws IllegalArgumentException Thrown if the given CharField is null.
	 */
	public CharField(CharField other) throws IllegalArgumentException
	{
		if (other == null)
		{
			throw new IllegalArgumentException("CharField object is null");
		}

		setInternal(new JNICharField((JNICharField) Field.getInternal(other)));
	}


	/**
	 * Returns the data stored within the CharField.
	 *
	 * @return A char value.
	 */
	public char getValue()
	{
		JNICharField jCharField = (JNICharField) getInternal();

		return jCharField.getValue();
	}
}
